src/ansible_compat/prerun.py

```diff
@@ -0,0 +1,427 @@
+        )
+        if run.returncode != 0:
+            _logger.error(run.stdout)
+            raise RuntimeError(run.returncode)
+
+
+def prepare_environment(required_collections: Optional[Dict[str, str]] = None) -> None:
+    """Make dependencies available if needed."""
+    if not options.configured:
+        # Allow method to be used without calling the command line, so we can
+        # reuse it in other tools, like molecule.
+        # pylint: disable=import-outside-toplevel,cyclic-import
+        # from ansiblelint.__main__ import initialize_options
+
+        # initialize_options()
+        pass
+
+    if not options.offline:
+        install_requirements("requirements.yml")
+        for req in pathlib.Path(".").glob("molecule/*/requirements.yml"):
+            install_requirements(str(req))
+
+    if required_collections:
+        for name, min_version in required_collections.items():
+            install_collection(
+                f"{name}:>={min_version}",
+                destination=f"{options.cache_dir}/collections"
+                if options.cache_dir
+                else None,
+            )
+
+    _install_galaxy_role()
+    # _perform_mockings()
+    _prepare_ansible_paths()
+
+
+def _get_galaxy_role_ns(galaxy_infos: Dict[str, Any]) -> str:
+    """Compute role namespace from meta/main.yml, including trailing dot."""
+    role_namespace = galaxy_infos.get('namespace', "")
+    if len(role_namespace) == 0:
+        role_namespace = galaxy_infos.get('author', "")
+    # if there's a space in the name space, it's likely author name
+    # and not the galaxy login, so act as if there was no namespace
+    if re.match(r"^\w+ \w+", role_namespace):
+        role_namespace = ""
+    else:
+        role_namespace = f"{role_namespace}."
+    if not isinstance(role_namespace, str):
+        raise RuntimeError("Role namespace must be string, not %s" % role_namespace)
+    return role_namespace
+
+
+def _get_galaxy_role_name(galaxy_infos: Dict[str, Any]) -> str:
+    """Compute role name from meta/main.yml."""
+    return galaxy_infos.get('role_name', "")
+
+
+def _get_role_fqrn(galaxy_infos: Dict[str, Any]) -> str:
+    """Compute role fqrn."""
+    role_namespace = _get_galaxy_role_ns(galaxy_infos)
+    role_name = _get_galaxy_role_name(galaxy_infos)
+    if len(role_name) == 0:
+        role_name = pathlib.Path(".").absolute().name
+        role_name = re.sub(r'(ansible-|ansible-role-)', '', role_name)
+
+    return f"{role_namespace}{role_name}"
+
+
+def _install_galaxy_role() -> None:
+    """Detect standalone galaxy role and installs it."""
+    if not os.path.exists("meta/main.yml"):
+        return
+    yaml = yaml_from_file("meta/main.yml")
+    if 'galaxy_info' not in yaml:
+        return
+
+    fqrn = _get_role_fqrn(yaml['galaxy_info'])
+
+    if 'role-name' not in options.skip_list:
+        if not re.match(r"[a-z0-9][a-z0-9_]+\.[a-z][a-z0-9_]+$", fqrn):
+            msg = (
+                """\
```

Comment:
I'd be nice to put this huge thing into a module-level constant as a template and render it here via `str.format()`.